File: services/backend/src/resources/eakte/service/models/eakten.go
package models

import (
	"time"

	stateModels "github.com/dcssoftware/bafoeg-manager/src/resources/eakte/states/models"
	"github.com/google/uuid"
)

type EakteModel struct {
	ID              uuid.UUID
	Aktenzeichen    string
	Typ             EakteModelType
	Vertraulichkeit stateModels.VertraulichkeitEnum
	Created         *time.Time
}

type EakteModelType struct {
	ID         uuid.UUID
	Name       string
	Identifier string
}

File: services/frontend/src/lib/api/eakten/types/eakten.ts
export interface EakteHttpResponse {
  maxCount: number;
  count: number;
  eakten: EakteModel[];
}

export interface EakteModel {
  id: string;
  aktenzeichen: string;
  type: EakteModelType;
  vertraulichkeit: string;
  created: Date;
}

export interface EakteModelType {
  id: string;
  name: string;
  identifier: string;
}